cmd/ci-secret-generator/main.go

```diff
@@ -120,17 +120,36 @@ func (o *options) validateConfig() error {
 }
 
 func executeCommand(command string) ([]byte, error) {
-	out, err := exec.Command("bash", "-o", "errexit", "-o", "nounset", "-o", "pipefail", "-c", command).CombinedOutput()
-	if err != nil {
-		return nil, fmt.Errorf("%s : %w", string(out), err)
+	cmd := exec.Command("bash", "-o", "errexit", "-o", "nounset", "-o", "pipefail", "-c", command)
+	var outBuf, errBuf bytes.Buffer
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &errBuf
+	if err := cmd.Run(); err != nil {
+		// The command completed with non zero exit code, standard streams *should* be available.
+		if _, ok := err.(*exec.ExitError); ok {
```

Comment:
Is this test following some standard recommendation?  From my reading of the code, it is possible to have another type of error and at least partial content in `Stdout`/`Stderr` (e.g. a `read` error in the middle of the execution).

Reply:
According to the `cmd.Run` documentation, the error of type `(*exec.ExitError)` is returned when the process being invoked did not complete successfully and, as far I have understood, it translates into what follows:

- the process completed
- its exit code differs from 0
- `cmd.Run` was able to both check the exit code and copy std streams into buffers

The best idiomatic, maybe yet outdated, approach I managed to find is [example_test.go](https://go-review.googlesource.com/c/go/+/213337/1/src/os/exec/example_test.go)

Do we care catching any execution path where `stderr` and/or `stdout` are not available?
IMHO, we should be fine handling the case we are sure they are available and just going past the others.

Reply:
My comment was specifically about:

```
// At this point we can't easily tell why the command failed,
// there is no guarantee neither stdout nor stderr are valid.
```

`outBuf` and `errBuf` are just byte arrays, they are always valid.  They may contain incomplete data in case the error was caused by the copies (and we can indicate that in the error message), but there is no reason to not display them.

Reply:
It sounds fine to me, I would propose something like:
```diff
failed to $action command "$command": $err
- output:
+ output (may be incomplete):
$stdout
- error output:
+ error output (may be incomplete):
$stderr
```